src/engine/qcommon/q_shared.h

```diff
@@ -1645,7 +1645,7 @@ inline float DotProduct( const vec3_t x, const vec3_t y )
 //
 #define MAX_CLIENTS         64 // JPW NERVE back to q3ta default was 128    // absolute limit
 
-#define GENTITYNUM_BITS     10 // JPW NERVE put q3ta default back for testing // don't need to send any more
+#define GENTITYNUM_BITS     16 // This is a huge number, typically we don't need more than 10 (1024 entities)
```

Comment:
I would say that

1. It's really hard for a player to add many entities. I spent half an hour with cheats and I could barely hit 1024 entities by building spikers. I don't think we are going close to that number without someone doing interesting stuff like computer-generated layouts — in which case if they are trying something crazy they probably know that already, and can't expect it to work well. In any case you need cheats to be able to build 100+ buildables.
2. Players can't create entities without placing buildables, throwing grenates, sending events or lighting fires. If this thing is generating too many entities then maybe this behavior shouldn't create so many of them in the first place? I would agree that there is an argument to be made against the excessive reusal of entities in game at the moment. For example to send events or damage indicators.
3. It seems that we have serious performance problems with even just 100 spikers, or [15 rocket pods](https://github.com/Unvanquished/Unvanquished/issues/535) anyway, so if we want to restrict buildable counts, we should do that explicitely.
4. In all case, it's not like we are handling an excess number of entities gracefully at the moment. It's just straight up killing sgame and disconnecting clients without telling why.

I think this may cause a problem for network message size though, like in https://github.com/Unvanquished/Unvanquished/issues/573.

Reply:
Note that I don't think there is something inherently blocking about the max message size `MAX_MSGLEN`. The message are sent in fragments over udp anyway, so this just means sending a few more fragments.

I think there may be one annoying thing though, this means that the arrays of size MAX_GENTITIES will need roughly 64 times more shared memory, even if the entities may not be needed. For the most sizeable of those, g_entities, this means 188MiB of memory, and that's arguably a lot.

Maybe 2^13 entities would be more reasonable ?

Reply:
I think 12 or 13 are reasonable. I think the best way to vet this is to play kai's map with max bots and spam all the firebombs you can. If you can't trigger it like that, then we have raised the entity limit enough 😄

Reply:
Updated